core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,998 @@
+      (Kokkos::Impl::always_true<I0, I1, I2, I3, I4, I5, I6, Is...>::value &&
+       (7 == rank)),
+      reference_type>
+  access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, I5 i5, I6 i6, Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4, i5, i6);
+  }
+
+  //------------------------------
+  // Rank 8
+
+  template <typename I0, typename I1, typename I2, typename I3, typename I4,
+            typename I5, typename I6, typename I7, typename... Is>
+  KOKKOS_FORCEINLINE_FUNCTION
+      std::enable_if_t<(Kokkos::Impl::always_true<I0, I1, I2, I3, I4, I5, I6,
+                                                  I7, Is...>::value &&
+                        (8 == rank)),
+                       reference_type>
+      access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, I5 i5, I6 i6, I7 i7,
+             Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4, i5, i6, i7);
+  }
+
+  //----------------------------------------
+  // Standard destructor, constructors, and assignment operators
+
+  KOKKOS_DEFAULTED_FUNCTION
+  ~View() = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View() = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View(const View& other) = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View(View&& other) = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View& operator=(const View& other) = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View& operator=(View&& other) = default;
+
+  KOKKOS_FUNCTION
+  View(typename base_t::data_handle_type p,
+       const typename base_t::mapping_type& m)
+      : base_t(p, m){};
+
+  //----------------------------------------
+  // Compatible view copy constructor and assignment
+  // may assign unmanaged from managed.
+
+  template <class OtherT, class... OtherArgs>
+  //    requires(std::is_constructible_v<
+  //             mdspan_type, typename View<OtherT, OtherArgs...>::mdspan_type>)
+  KOKKOS_INLINE_FUNCTION View(
+      const View<OtherT, OtherArgs...>& other,
+      std::enable_if_t<
+          std::is_constructible_v<
+              mdspan_type, typename View<OtherT, OtherArgs...>::mdspan_type>,
+          void*> = nullptr)
+      : base_t(static_cast<typename mdspan_type::data_handle_type>(
+                   other.data_handle()),
+               static_cast<typename mdspan_type::mapping_type>(other.mapping()),
+               static_cast<typename mdspan_type::accessor_type>(
+                   other.accessor())) {
+    base_t::check_basic_view_constructibility(other.mapping());
+  }
+
+  //----------------------------------------
+  // Compatible subview constructor
+  // may assign unmanaged from managed.
+
+  template <class RT, class... RP, class Arg0, class... Args>
+  KOKKOS_INLINE_FUNCTION View(const View<RT, RP...>& src_view, const Arg0 arg0,
+                              Args... args)
+      : base_t(Impl::subview_ctor_tag, src_view, arg0, args...) {}
```

Comment:
We might want to constrain this more, see https://github.com/kokkos/kokkos/issues/7735, or even deprecate public access.